### PR TITLE
Dataset client signature (part 1)

### PR DESCRIPTION
**Pull request by RicoGit**

### CODE REVIEW

File: contract/core/src/main/scala/fluence/contract/ops/ContractRead.scala [hunk 1]
```diff
@@ -106,35 +124,32 @@ object ContractRead {
...
+      } yield ()
 
     /**
      * Checks that signature matches contract's offer
      *
      * @param signature Signature to check
      * @param checker Signature checker
      */
-    def checkOfferSignature[F[_]: Monad](
+    private def checkOfferSignature[F[_]: Monad](
```
Comment (alari) [inline contract/core/src/main/scala/fluence/contract/ops/ContractRead.scala]: What is the reason for making it private?

File: contract/core/src/main/scala/fluence/contract/ops/ContractRead.scala [hunk 2]
```diff
@@ -143,64 +158,124 @@ object ContractRead {
...
-        allParticipants ← checkAllParticipants()
-        participantSealResult ← participantsSeal.filter(seal ⇒ Key.checkPublicKey(id, seal.publicKey)) match {
-          case Some(sign) ⇒ checker.check[F](sign, getParticipantsBytes).map(_ ⇒ true)
-          case None ⇒ EitherT.rightT[F, CryptoErr](false)
+        _ ← checkOfferSeal()
+        _ ← checkExecStateSeal()
+        _ ← checkAllParticipants()
+        participantSealResult ← checkParticipantsSeal
+      } yield
+        participantSealResult match {
```
Comment (alari) [inline contract/core/src/main/scala/fluence/contract/ops/ContractRead.scala]: It's the same as `.isDefined`

File: contract/core/src/main/scala/fluence/contract/ops/ContractRead.scala [hunk 3]
```diff
@@ -143,64 +158,124 @@ object ContractRead {
      * @return Whether this contract is a valid blank offer (with no participants, with client's signature)
      */
     def isBlankOffer[F[_]: Monad]()(implicit checker: SignatureChecker): EitherT[F, CryptoErr, Boolean] =
-      if (participants.isEmpty) checkOfferSeal() else EitherT.rightT(false)
-
-    /**
-     * @return Whether this contract offer was signed by a single node and client, but participants list is not sealed yet
-     */
-    def isSignedParticipant[F[_]: Monad]()(implicit checker: SignatureChecker): EitherT[F, CryptoErr, Boolean] =
```
Comment (alari) [inline contract/core/src/main/scala/fluence/contract/ops/ContractRead.scala]: Where has it gone? Can't find out from the PR what is used instead for gathering contract offer signatures on client side

File: build.sbt [hunk 4]
```diff
@@ -440,7 +440,12 @@ lazy val `contract-core` = crossProject(JVMPlatform, JSPlatform)
   .withoutSuffixFor(JVMPlatform)
   .crossType(FluenceCrossType)
   .in(file("contract/core"))
-  .settings(commons)
+  .settings(
+    commons,
+    libraryDependencies ++= Seq(
+      scalatest
```
[the remark this answers is not in the document]
Comment (RicoGit) [reply build.sbt]: it is, unfortunately only jvm tests can be runned from idea with debugger. I moved tests to crosplatform module.

File: contract/core/src/main/scala/fluence/contract/ops/ContractRead.scala (hunk 1 above)
Comment (RicoGit) [reply contract/core/src/main/scala/fluence/contract/ops/ContractRead.scala]: This methods isn't used anywhere, why it should be public? This functionality isn't needed outside of contract, for all cases similar there is checkOfferSeal method

File: contract/core/src/main/scala/fluence/contract/ops/ContractRead.scala (hunk 2 above)
Comment (RicoGit) [reply contract/core/src/main/scala/fluence/contract/ops/ContractRead.scala]: sure, thanks

File: contract/core/src/main/scala/fluence/contract/ops/ContractRead.scala (hunk 3 above)
Comment (RicoGit) [reply contract/core/src/main/scala/fluence/contract/ops/ContractRead.scala]: This methods wasn't used anywhere, and it seemed to me useless. participantSigned methods is used  instead

**Pull request merged**